### PR TITLE
Feature portal38 woadd 648

### PR DESCRIPTION
**Pull request by deguzman22**
*** PLEASE DO NOT OPEN PULL REQUESTS VIA GITHUB ***

The moodle.git repository at Github is just a mirror of the official repository. We do not accept pull requests at Github.

See CONTRIBUTING.txt guidelines for how to contribute patches for Moodle. Thank you.

--

### CODE REVIEW

File: blocks/gu_spoverview/templates/spoverview.mustache [hunk 1]
```diff
@@ -23,22 +23,22 @@
     <div class="row">
         <div class="assessments-item assessments-submitted col-md-3 col-sm-6 col-xs-12">
             <img src="{{assessments_submitted_icon}}" alt="{{assessments_submitted_str}}" class="assessments-item-icon"/>
-            <h4 class="assessments-item-count color-primary">{{assessments_submitted}}</h4>
+            <h4 class="assessments-item-count color-primary submitted-assignment-count">{{assessments_submitted}}</h4>
```
Comment (francolouietm) [inline blocks/gu_spoverview/templates/spoverview.mustache]: Remove added classes.

File: blocks/gu_spoverview/block_gu_spoverview.php [hunk 2]
```diff
@@ -81,6 +81,7 @@ public function get_content() {
             }
         }
 
+
```
Comment (francolouietm) [inline blocks/gu_spoverview/block_gu_spoverview.php:84]: Remove added line.

**Review by francolouietm — CHANGES_REQUESTED** [review]

File: blocks/gu_spoverview/templates/spoverview.mustache (hunk 1 above)
Comment (deguzman22) [reply blocks/gu_spoverview/templates/spoverview.mustache]: noted

File: blocks/gu_spoverview/block_gu_spoverview.php (hunk 2 above)
Comment (francolouietm) [inline blocks/gu_spoverview/block_gu_spoverview.php:84]: Please remove this extra line.

**Review by francolouietm — CHANGES_REQUESTED** [review]

File: blocks/gu_spoverview/tests/spoverview_test.php [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+
+    /**
+     * Returns all user assignments including submission status and extension due date.
+     * 
+     * @param int $userid
+     * @param array $courseids
+     * @return stdClass Assignment objects if records are returned,
+     *  otherwise return empty object
+     */
+    public static function get_user_assignments($userid, $courseids) {
```
Comment (francolouietm) [inline blocks/gu_spoverview/tests/spoverview_test.php]: Remove this method as this already exists in block_gu_spoverview().

File: blocks/gu_spoverview/tests/spoverview_test.php [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    }
+
+    /**
+     * Returns count of graded assessments (activity modules)
+     * 
+     * @param int $userid
+     * @param array $courseids
+     * @return int Count of Assessment records
+     */
+    public static function get_user_assessments_count($userid, $courseids) {
```
Comment (francolouietm) [inline blocks/gu_spoverview/tests/spoverview_test.php]: Remove this method as this already exists in block_gu_spoverview().

File: blocks/gu_spoverview/tests/spoverview_test.php [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+        $lang = 'block_gu_spoverview';
+
+        $assignments_submitted = 0;
+        $assignments_tosubmit = 0;
+        $assignments_overdue = 0;
+        
+        $courses = enrol_get_all_users_courses($this->student->id, true);
+        $courseids = array_column($courses, 'id');
+
+        $assignments = (count($courseids) > 0) ? self::get_user_assignments($this->student->id, $courseids) : 0;
```
Comment (francolouietm) [inline blocks/gu_spoverview/tests/spoverview_test.php]: Add $this->spoverview = new block_gu_spoverview(); and use $this->spoverview to call get_user_assignments.

File: blocks/gu_spoverview/tests/spoverview_test.php [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+        $assessments_marked = (count($courseids) > 0) ? self::get_user_assessments_count($this->student->id, $courseids) : 0;
```
Comment (francolouietm) [inline blocks/gu_spoverview/tests/spoverview_test.php]: Add $this->spoverview = new block_gu_spoverview(); and use $this->spoverview to call get_user_assessments_count.

**Review by francolouietm — CHANGES_REQUESTED** [review]

File: blocks/gu_spoverview/tests/spoverview_test.php [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+                                    'category' => $this->category->id
+                                )
+                        );
+        $gen->enrol_user($this->student->id, $this->course->id, 'student');
+        $gen->enrol_user($this->teacher->id, $this->course->id, 'editingteacher');
+
+        $gen->create_module('assign', array(
+            'course' => $this->course->id,
+            'duedate' => time() * 30 * 30,
+            'status' => 'new'
```
Comment (francolouietm) [inline blocks/gu_spoverview/tests/spoverview_test.php]: 'assign' module does not have status column. Status is found in assign_submission. Please create an insert_record('assign_submission') instead.

File: blocks/gu_spoverview/tests/spoverview_test.php [hunk 8]
```diff
@@ -0,0 +1,251 @@
...
+        ));
+
+        $gen->create_module('assign', array(
+            'course' => $this->course->id,
+            'duedate' => time() - 10,
+            'status' => 'new'
```
Comment (francolouietm) [inline blocks/gu_spoverview/tests/spoverview_test.php:64]: 'assign' module does not have status column. Status is found in assign_submission. Please create an insert_record('assign_submission') instead.

File: blocks/gu_spoverview/tests/spoverview_test.php [hunk 9]
```diff
@@ -0,0 +1,251 @@
...
+        ));
+
+        $gen->create_module('assign', array(
+            'course' => $this->course->id,
+            'duedate' => time() - 20,
+            'status' => 'new'
```
Comment (francolouietm) [inline blocks/gu_spoverview/tests/spoverview_test.php:71]: 'assign' module does not have status column. Status is found in assign_submission. Please create an insert_record('assign_submission') instead.

File: blocks/gu_spoverview/tests/spoverview_test.php [hunk 10]
```diff
@@ -0,0 +1,251 @@
...
+        ));
+
+        $gen->create_module('assign', array(
+            'course' => $this->course->id,
+            'duedate' => time() - 20,
+            'status' => 'new'
```
Comment (francolouietm) [inline blocks/gu_spoverview/tests/spoverview_test.php]: 'assign' module does not have status column. Status is found in assign_submission. Please create an insert_record('assign_submission') instead.

**Review by francolouietm — CHANGES_REQUESTED** [review]

**Review by francolouietm — APPROVED** [review]

**Pull request closed without merging**